tendrl-dashboard.spec

```diff
@@ -8,16 +8,57 @@ Group:   Applications/System
 Source0: %{name}-%{version}.tar.gz
 URL: https://github.com/Tendrl/dashboard
 
-Requires: tendrl-api
+BuildRequires: npm
+BuildRequires: nodejs-packaging
+BuildRequires: fontconfig
 
 %description
 Contains the JavaScript GUI content for the Tendrl front-end components
 (dashboard, login screens, administration screens)
 
 %prep
-%autosetup
+%setup -q -n %{name}-%{version}
+%nodejs_fixdep foomodule
 
 %build
+npm install gulp
```

Comment:
Are you suggesting that they need to package (or bundle in worse case) hundreds of nmp packages to make this work in koji? 

I'm asking because I want to understand what is the best of all those bad options we have here.
And I understand that this change is not be the proper way which could pass fedora packaging review.

Right now, the build is happening outside of rpmbuild and source tarball contains the result of the build, not the source. So this change is kind of an improvement from this point of view (at least for me). Now the question is if approach presented here (running build during rpmbuild while installing all required deps via npm) is good enough in the long term. Which obviously isn't (at least from fedora packaging policy view).

During typing this reply, one option come to my mind though: could we create package tendlr-dashboard-build package which would bundle all npm dependencies required for building tendlr-dashboard, so that tendlr-dashboard itlself would be clean (and could be requild fine without problems), as all the ugly things (bundling all those npm stuff) would be done in tendrl-dashboard-build. The point is that we  don't need to ship those build requirements for the dashboard to work, those are just build time deps.

It's ugly as well, but it would be at least improvement compared to the current state.